### PR TITLE
feat(wizards/reportcontrol): add basic report manipulation capability

### LINKED ISSUE #144: Report control block manipulation capability
As a user of OpenSCD I want to manage `IED` configuration. I want create/delete and update `Report` control blocks. 

**Requirement**:
- TBD

**Type Definition**:
- TBD

**Restriction** :
- TBD

**Edition 1 vs Edition 2**
- TBD

### PR DESCRIPTION
**Pull request by JakobVogelsang**
Closes #144

### CODE REVIEW

File: src/zeroline/foundation.ts [hunk 1]
```diff
@@ -104,6 +104,26 @@ export type ElementEditor = Element & {
   element: Element;
 };
 
+export function updateNamingAction(element: Element): WizardActor {
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply src/zeroline/foundation.ts]: There is no. We use `cloneElement` in other update functions to create the new element for the update action.

File: test/integration/zeroline-pane.test.ts [hunk 2]
```diff
@@ -43,6 +42,19 @@ describe('zeroline-pane wizarding editing integration', () => {
     );
   });
 
+  it('opens selectReportControlWizard for the complete SCL file', async () => {
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply test/integration/zeroline-pane.test.ts:45]: We do that in all unit tests. You are right, and there are cases where the function name and the test case do not match anymore for this exact reason. 

I would move that out of this PR however and change that into a separate story. #316

File: test/integration/zeroline-pane.test.ts [hunk 3]
```diff
@@ -43,6 +42,19 @@ describe('zeroline-pane wizarding editing integration', () => {
...
+    zeroline.reportcontrol.click();
+    await parent.updateComplete;
+    expect(parent.wizardUI.dialog).to.exist;
+    const reportControlList = <FilteredList>(
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply test/integration/zeroline-pane.test.ts:49]: The description is misleading, I would say. The test checks that the wizard show the correct amount of `ReportControl`'s.

[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply test/integration/zeroline-pane.test.ts:49]: The descrition is missleading it think. The test checks that not only the `ReportControl` of one IED is shows but all. There is a difference if this wizard has been triggered form `ied-editor` or `zeroline-pane`.

File: test/unit/wizards/trgops.test.ts [hunk 4]
```diff
@@ -79,7 +79,7 @@ describe('trgops wizards', () => {
       await input.requestUpdate();
       const editorAction = updateTrgOpsAction(trgOps);
       const actions = editorAction(inputs, newWizard());
-      expect(actions.length).to.equal(1);
+      expect(actions.length).to.equal(2);
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply test/unit/wizards/trgops.test.ts:82]: Sorry for that, the commit history for this PR is not good :(. Trigger options might influence the attribute `intPd` in the `ReportControl`. This is the case with this particular test.

File: test/integration/wizards/reportcontrol-wizarding-editing-integration.test.ts [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+        element.wizardUI.dialog?.querySelector('wizard-textfield[label="name"]')
+      );
+      await nameField.updateComplete;
+      expect(nameField.value).to.equal(
+        doc.querySelectorAll('ReportControl')[0].getAttribute('name')
+      );
+    });
+  });
+
+  describe('editGseControlWizard', () => {
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply test/integration/wizards/reportcontrol-wizarding-editing-integration.test.ts:59]: This is. Thank you!

File: test/integration/wizards/reportcontrol-wizarding-editing-integration.test.ts [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+        )!;
+        primaryAction = <HTMLElement>(
+          element.wizardUI.dialog?.querySelector(
+            'mwc-button[slot="primaryAction"]'
+          )
+        );
+        await nameField.updateComplete;
+      });
+
+      it('edits name attribute on primary action', async () => {
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply test/integration/wizards/reportcontrol-wizarding-editing-integration.test.ts:83]: This is incorrect, you are right.

File: test/integration/wizards/reportcontrol-wizarding-editing-integration.test.ts [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+        await editDataSetButton.updateComplete;
+        editDataSetButton.click();
+        await new Promise(resolve => setTimeout(resolve, 100)); // await animation
+        const nameField = <WizardTextField>(
+          element.wizardUI.dialog?.querySelector(
+            'wizard-textfield[label="name"]'
+          )
+        );
+        await nameField.updateComplete;
+        expect(nameField.value).to.equal(
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply test/integration/wizards/reportcontrol-wizarding-editing-integration.test.ts:104]: That is true. Thank you!

File: test/integration/zeroline/iededitor-wizarding-editing-integration.test.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    const gseControlList = <FilteredList>(
+      parent.wizardUI.dialog?.querySelector('filtered-list')
+    );
+    await gseControlList.updateComplete;
+    expect(gseControlList.items.length).to.equal(
+      doc.querySelectorAll('IED[name="IED1"] GSEControl').length
+    );
+  });
+
+  it('opens selectReportControlWizard for the complete SCL file', async () => {
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply test/integration/zeroline/iededitor-wizarding-editing-integration.test.ts:42]: Yes.

File: test/integration/wizards/optfields-wizarding-editing.test.ts [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+      expect(reportCb.querySelector('OptFields')).to.have.attribute(
+        'dataSet',
+        'false'
+      );
+    });
+    it('returns to editReportControlWizard after update', async () => {
+      dataSetField.value = 'false';
+      primaryAction.click();
+      await element.updateComplete;
+      await new Promise(resolve => setTimeout(resolve, 100)); // await animation
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply test/integration/wizards/optfields-wizarding-editing.test.ts:59]: Thanks for the hint. This is indeed slowing the test down, but we have not found a solution, yet.

File: test/integration/wizards/reportcontrol-wizarding-editing-integration.test.ts [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+        reportControl = doc.querySelector(
+          'LN[lnClass="XSWI"] ReportControl[name="ReportCb"]'
+        )!;
+        const wizard = editReportControlWizard(reportControl);
+        element.workflow.push(wizard);
+        await element.requestUpdate();
+        nameField = element.wizardUI.dialog!.querySelector(
+          'wizard-textfield[label="name"]'
+        )!;
+        primaryAction = <HTMLElement>(
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply test/integration/wizards/reportcontrol-wizarding-editing-integration.test.ts:75]: I think this is also a debate for test refactoring as we use this pattern everywhere in the tests we should update everywhere as well. Thank you for the hint.

File: src/zeroline/foundation.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (b-ono) [reply src/zeroline/foundation.ts]: Yes, but the commit message is "change cloneElement..." and the only change in the commit itself is the introduction of this new function

[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply src/zeroline/foundation.ts]: Oh now I understand :), sorry. Yes, this true, I added the wrong file to the commit. We stash all commits at the end and at least in my case the commit in the PR are not always clean and small. I need to that better :).

**Pull request closed without merging**